charts/cass-operator/templates/rolebinding.yaml

```diff
@@ -11,7 +10,8 @@ roleRef:
   kind: ClusterRole
   name: {{ template "k8ssandra-common.fullname" . }}
   apiGroup: rbac.authorization.k8s.io
-  {{- else }}
+{{- if not .Values.global.clusterScoped }}
```

Comment:
Thanks for this contribution @mlallaouret, from what I can understand here, it looks like we need a ClusterRole and ClusterRoleBinding irrespective of whether `.Values.global.clusterScoped` is true. 

It appears that this bit of logic might also create a RoleBinding to the same ClusterRole, but only when  `.Values.global.clusterScoped` is false. From my understanding, the RoleBinding should probably never be created, as I think the ClusterRoleBinding will be sufficient to give bind the ServiceAccount to the ClusterRole, and it is getting created in either case.

Is my understanding correct here? Can we remove the RoleBinding entirely?

Reply:
Hi Miles !
Jumping in because I reviewed Marc's PR before we submitted it to you ;) 

For me, the `RoleBinding` below targets the (conditionally created) `Role` and not the (unconditionally created) `ClusterRole` in its `roleRef`

To sum up :
- if `.Values.global.clusterScoped` is `true`, we have those RBAC resources
  - a `ClusterRole` with full permissions for the operator (cluster-scoped & namespaced-scopes resources )
  - a `ClusterRoleBinding` targeting the `ClusterRole` above

- if `.Values.global.clusterScoped` is `false` (the default for the chart), we have those RBAC resources
  - a `ClusterRole` with minimalistic permissions to access cluster-scoped resources (storageclasses & persistentvolumes)
  - a `ClusterRoleBinding` targeting the `ClusterRole` above
  - a `Role` with full permissions for the operator except those for cluster-scoped resources (storageclasses & persistentvolumes)
  - a `RoleBinding` targeting the `Role` above

If I understand well, the reason behind this is that you can't give access to cluster-scoped resources with a `Role`, only with a `ClusterRole`

Reply:
What @KeepItSimpleStupid said is right. However, these changes in this repo would be overwritten as these are generated files.

So the updates should happen in https://github.com/k8ssandra/cass-operator/blob/master/scripts/release-helm-chart.sh and then rerun here.

I can get back to this subject next week.

Reply:
Thanks Michael !

Am I wrong if I say that the script you pointed changes only :
- `charts/cass-operator/crds` folder
- `charts/cass-operator/templates/role.yaml`
- `charts/cass-operator/Chart.yaml` (to bump appVersion)
- `charts/cass-operator/values.yaml`

So touching `charts/cass-operator/templates/rolebinding.yaml` in this PR should be safe ?

Reply:
Right, so it seems the rolebinding.yaml wasn't regenerated. Not that it needs to be. So the PR should be safe, yep.